core/src/main/scala/org/http4s/Uri.scala

```diff
@@ -193,6 +193,57 @@ object Uri extends UriFunctions {
       }
   }
 
+  final class Port private (val toInt: Int) extends AnyVal with Ordered[Port] {
```

Comment:
This one is more useful even outside a `URI` and should maybe move to the top level.  It wanted to be part of our `Host` header, and could be used in various places on the `Request` as well.  And then `ServerBuilder` would also be a natural fit.